### DISCUSSION TITLE
Installing tools from the web or GitHub Releases

### DISCUSSION THREAD

**Discussion by mgnslndh** [cake-build/cake#4309] [General]
1. Is there a tool provider that can download the tool from the web? 
2. Is there a tool provider that can download the tool from GitHub Releases?

**Comment by devlead** [comment]
Currently there's built in support for fetching tools from NuGet feeds both as a package or .NET tool, there're also third party modules for i.e Chocolatey
https://www.nuget.org/packages/Cake.Chocolatey.Module

I'm not aware of an module to fetch tools from github releases, do you have an example tool available as a GitHub release you want to use with Cake?

**Reply by mgnslndh** [reply]
I was thinking about [cyclonedx-cli]( https://github.com/CycloneDX/cyclonedx-cli ) but I guess it is probably just as easy to download it and publish a custom NuGet package on the internal feed.

**Reply by devlead** [reply]
It's available on NuGet, so you should be able to do

```cake
#tool dotnet:?package=CycloneDX&version=3.0.6
```

**Reply by mgnslndh** [reply]
I think those are different tools. The one you are referring to does not seem to support converting from SPDX to CycloneDX.

But the conversion caused other problems for me so in order to proceed I'll just use Covenant to produce the CycloneDX SBOM and skip the conversion process.
